pyclient/pydeephaven/session.py

```diff
@@ -0,0 +1,289 @@
+#
+#  Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+
+import os
+import threading
+# from concurrent.futures import ThreadPoolExecutor
+from typing import List
+
+import pyarrow
+from bitstring import BitArray
+
+from pydeephaven._arrow_flight_service import ArrowFlightService
+from pydeephaven._console_service import ConsoleService
+from pydeephaven._session_service import SessionService
+from pydeephaven._table_service import TableService
+from pydeephaven.dherror import DHError
+from pydeephaven.proto import ticket_pb2
+from pydeephaven.query import Query
+from pydeephaven.table import Table, TimeTable, EmptyTable
+
+
+class Session:
+    """ A Session object represents a connection to the Deephaven data server. It contains a number of convenience
+    methods for asking the server to create tables, import Arrow data into tables, merge tables, run Python scripts, and
+    execute queries.
+
+    Attributes:
+        tables (list[str]): names of the tables available in the server after running scripts
+        is_alive (bool): check if the session is still alive (may refresh the session)
+    """
+
+    def __init__(self, host: str = 'localhost', port: int = 10000):
+        """ Initialize a Session object that connects to the Deephaven server
+        Args:
+            host (str): the host name or IP address of the remote machine, default is 'localhost'
+            port (int): the port number that Deephaven server is listening on, default is 10000
+
+        Returns:
+            None
+
+        Raises:
+            DHError
+
+        """
+        self._r_lock = threading.RLock()
+        self._last_ticket = 0
+        self._ticket_bitarray = BitArray(1024)
+        self.host = os.environ.get("DHCE_HOST", host)
+        self.port = os.environ.get("DHCE_PORT", port)
```

Comment:
I think we want a different order of precedence:

1. passed in value
2. environment value
3. default value

As it stands, the order is:

1. environment value
2. passed in value
3. default value

Reply:
yes, I agree. will change.